nshlib/nsh_altconsole.c

```diff
@@ -292,12 +294,6 @@ int nsh_consolemain(int argc, char *argv[])
   netinit_bringup();
 #endif
 
-#if defined(CONFIG_NSH_ARCHINIT) && defined(CONFIG_BOARDCTL_FINALINIT)
-  /* Perform architecture-specific final-initialization (if configured) */
-
-  boardctl(BOARDIOC_FINALINIT, 0);
-#endif
-
```

Comment:
I don't understand why this is removed.  It seems unrelated to the PR.

Reply:
BOARDIOC_FINALINIT spread in many place, it's better to put into one place: nsh_initscript.
Since this PR contain all issues I found in reviewing nsh iniitialization process, it's better to review each patch one by one at here:
https://github.com/apache/incubator-nuttx-apps/pull/185/commits
Each commit message describe why I make this change.